### ISSUE TITLE
Cop idea: Predicate Methods Suffix

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12360]
## Description

The names of predicate methods (methods that return a boolean value) should end in a question mark (i.e. `Array#empty?`). Methods that don't return a boolean, shouldn't end in a question mark.

https://rubystyle.guide/#bool-methods-qmark

## Describe the solution you'd like

```ruby
# bad
def my_method
  x == y
end

# good
def my_method?
  x == y
end
```

```ruby
# bad
def my_method
  return false if x == y

  true
end

# good
def my_method?
  return false if x == y

  true
end
```

```ruby
# good
def my_method
  return false if x == y

  nil
end
```

**Comment by fatkodima** [comment]
`Naming/PredicateName` ?

**Comment by ydakuka** [comment]
> Naming/PredicateName ?

As far as I know, this cop does not check method body, but I propose a cop which should check this.
